### PR TITLE
IMPROVE platform indepence by removing direct dependency to fcntl

### PR DESCRIPTION
**Pull request by patkan**
- use portalocker instead
- portalocker uses on posix-systems fcntl
- on nt it does win32api-calls
  @EmJay276 please test this on windows

### CODE REVIEW

File: FabLabKasse/libs/portalocker/portalocker.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+For the 'nt' platform, this module requires the Python Extensions for Windows.
+Be aware that this may not work as expected on Windows 95/98/ME.
+
+History:
+
+I learned the win32 technique for locking files from sample code
+provided by John Nielsen <nielsenjf@my-deja.com> in the documentation
+that accompanies the win32 modules.
+
+License: PSFL
```
Comment (mgmax) [inline FabLabKasse/libs/portalocker/portalocker.py]: we should include the PSFL license text

**Comment by patkan** [comment]
The best way would be to use https://github.com/WoLpH/portalocker as submodule or the corresponding pip-package.

**Comment by patkan** [comment]
I updated the project to use the pip-package.
Furthermore I tested the provisioning and it works.

**Comment by patkan** [comment]
@EmJay276 it would be really nice if you could test it now under Windows. You will probably run into problems later on, but at least file locking should work now ;-)

**Comment by patkan** [comment]
Ok, we tested it together and the portalocker-lib currently seems to have problems with the win32con-lib. The maintainer @WoLpH seems to be working on it.
If no one sees any regressions when using this lib ( @cod3monk Does portalocker work on Mac?), I would propose merging this.

**Comment by wolph** [comment]
@patkan just a fyi, portalocker has support for the newer windows locking methods now but it seems there is an issue with shared locks: https://github.com/WoLpH/portalocker/issues/29

**Pull request merged**